src/WebJobs.Script.WebHost/WebScriptHostRequestManager.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Script.Binding.Http;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    public class WebScriptHostRequestManager : HttpRequestManager
+    {
+        private readonly HostPerformanceManager _performanceManager;
+        private readonly IMetricsLogger _metricsLogger;
+        private readonly int _performanceCheckPeriodSeconds;
+        private DateTime _lastPerformanceCheck;
+        private bool _rejectAllRequests;
+
+        public WebScriptHostRequestManager(HttpConfiguration config, HostPerformanceManager performanceManager, IMetricsLogger metricsLogger, TraceWriter traceWriter, int performanceCheckPeriodSeconds = 15) : base(config, traceWriter)
+        {
+            _performanceManager = performanceManager;
+            _metricsLogger = metricsLogger;
+            _performanceCheckPeriodSeconds = performanceCheckPeriodSeconds;
+        }
+
+        protected override bool RejectAllRequests()
```

Comment:
I'd like to see the counter that is exceeded logged for the user instead of the generic message.

Do we have control over how traffic routing, so we can avoid sending requests to high load instances?

I don't think this needs to be configurable, as in high load cases you're just going to hit a 502/503, so the 429 would always give you more actionable info.